lib/anvil.js

```diff
@@ -0,0 +1,182 @@
+const fixData = require('../data/repairable_items.json')
+const maxItemDura = require('../data/max_item_durability.json')
+const armorMapping = require('../data/armor_category_mapping.json')
+const anvilUses = require('../data/anvil_uses.json').regular
+
+function loader (mcData, Item) {
+  function combine (itemOne, itemTwo, creative, renamedName) {
+    // TODO implement renamings
+    const rename = typeof renamedName === 'string'
+    const data = {
+      finalEnchs: []
+    }
+    try {
+      if (!combinePossible(itemOne, itemTwo) && itemTwo !== null) return { xpCost: 0, invalid: true } // throws if not possible
+      let cost = baseCost(itemOne) + baseCost(itemTwo)
+      if (rename) {
+        const renameCost = getRenameCost(itemOne)
+        if (renameCost === -1) return { xpCost: 0, invalid: true }
+        cost += renameCost
+      }
+      if (itemOne.metadata !== 0) {
+        const { xpLevelCost: repairCost, fixedDurability, usedMats } = getRepairCost(itemOne, itemTwo)
+        data.fixedDurability = fixedDurability
+        data.usedMats = usedMats
+        cost += repairCost
+      }
+      if (itemTwo && (itemTwo.name === itemOne.name || itemTwo.name === 'enchanted_book')) {
+        const { xpLevelCost: enchantCost, finalEnchs } = combineEnchants(itemOne, itemTwo, creative)
+        data.finalEnchs = finalEnchs
+        if (enchantCost === 0 && !rename && itemOne.metadata === 0) return { xpCost: 0, invalid: true } // no change
+        cost += enchantCost
+      }
+      let finalItem = null // defaults to null for if first item is air, TODO: just be the original item with new name
+      if (itemOne) {
+        finalItem = new Item(itemOne.type, itemOne.count, itemOne.metadata + data.fixedDurability, null) // replace null with itemOne.nbt
```

Comment:
doesnt fixedDurability already include metadata ?

Reply:
yes, its defaults to 0, otherwise it'll be = to the number of durability pts restored.